### ISSUE TITLE
Upgrade Darwin package to build crystal against LLVM 17

### ISSUE THREAD

**Issue by miry** [crystal-lang/distribution-scripts#280]
Homebrew crystal uses llvm 17. The current solution for darwin using llvm 15.

Should it crystal package use the same LLVM version as for homebrew build script?

**Comment by straight-shoota** [comment]
Yeah, that would be ideal. I think the updates for darwin are usually lagging behind (or happen less frequently) because it's more effort with omnibus. It shouldn't really matter *that* much which LLVM version is used though. Maybe newer ones offer some small performance gains, but usually it shouldn't be much.

**Comment by straight-shoota** [comment]
Related: #186
